Bento/Bento/If.swift

```diff
@@ -1,99 +1,69 @@
-public struct If<T> {
-    let condition: () -> Bool
-    let generator: () -> T
-
-    public static func iff(_ condition: @autoclosure @escaping () -> Bool, _ generator: @autoclosure @escaping () -> T) -> If<T> {
-        return If(condition: condition, generator: generator)
-    }
-
-    public static func iff(_ condition: @autoclosure @escaping () -> Bool, _ generator: @escaping () -> T) -> If<T> {
-        return If(condition: condition, generator: generator)
+extension Optional {
```

Comment:
shouldn't we add type constraints here? `where Wrapped == <BentoTypes>` ?

Reply:
`Section` / `[Section]` / `Node` / `[Node]` will be `Wrapped` in Bento, and the implementation itself is very generic, so I think we should keep this generic form.

Reply:
ahhhh, nevermind. its not a public extension
👍

Reply:
the methods are public though, so it will be accessible for any optional, but as @inamiy said they are generic.